src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,250 @@
+            tasks.add(new Deadline(description, by));
+            taskCount++;
+            printAddTaskMessage(tasks.get(tasks.size() - 1));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+        }
+    }
+
+    public static void addEvent(String line) {
+        try {
+            int descriptionStart = 6;
+            int descriptionEnd = line.indexOf("/at") - 1;
+            String description = line.substring(descriptionStart, descriptionEnd);
+            int atStart = line.indexOf("/at") + 4;
+            String at = line.substring(atStart);
+            tasks.add(new Event(description, at));
+            taskCount++;
+            printAddTaskMessage(tasks.get(tasks.size() - 1));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The description of an event cannot be empty.");
+        }
+    }
+
+    public static void markDone(String line) {
+        try {
+            int itemNum = Integer.parseInt(line.substring(5));
+            tasks.get(itemNum - 1).setAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(tasks.get(itemNum - 1).toString());
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The index of the task to be marked as done cannot be empty.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The index of the task to be marked as done is out of range");
+        } catch (NumberFormatException e) {
+            System.out.println("OOPS!!! The index of the task to be marked as done is invalid");
+        }
+    }
+
+    public static void deleteTask(String line) {
+        try {
+            int itemNum = Integer.parseInt(line.substring(7));
+            Task toBeDeleted = tasks.get(itemNum - 1);
+            tasks.remove(itemNum - 1);
+            taskCount--;
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(toBeDeleted.toString());
+            System.out.println("Now you have " + tasks.size() + " tasks in the list");
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The index of the task to be deleted cannot be empty.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("OOPS!!! The index of the task to be deleted is out of range");
+        } catch (NumberFormatException e) {
+            System.out.println("OOPS!!! The index of the task to be deleted is invalid");
+        }
+
+    }
+
+
+    public static void request() {
+        String line;
+        Scanner in = new Scanner(System.in);
+
+        line = in.nextLine();
+        while (!line.equals("bye")) {
+            System.out.println("---------------------------------------------------------");
+            if (line.equals("list")) {
```

Comment:
maybe we should explore the use of constants here in order to improve the code readability and the need of magic strings?